// @docusaurus/theme-classic/lib/theme/BlogTagsListPage/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _Layout = _interopRequireDefault(require("@theme/Layout"));

var _Link = _interopRequireDefault(require("@docusaurus/Link"));

var _BlogSidebar = _interopRequireDefault(require("@theme/BlogSidebar"));

var _Translate = require("@docusaurus/Translate");

var _themeCommon = require("@docusaurus/theme-common");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
function getCategoryOfTag(tag) {
  // tag's category should be customizable
  return tag[0].toUpperCase();
}

function BlogTagsListPage(props) {
  const {
    tags,
    sidebar
  } = props;
  const title = (0, _Translate.translate)({
    id: 'theme.tags.tagsPageTitle',
    message: 'Tags',
    description: 'The title of the tag list page'
  });
  const tagCategories = {};
  Object.keys(tags).forEach(tag => {
    const category = getCategoryOfTag(tag);
    tagCategories[category] = tagCategories[category] || [];
    tagCategories[category].push(tag);
  });
  const tagsList = Object.entries(tagCategories).sort(([a], [b]) => a.localeCompare(b));
  const tagsSection = tagsList.map(([category, tagsForCategory]) => <article key={category}>
        <h2>{category}</h2>
        {tagsForCategory.map(tag => <_Link.default className="padding-right--md" href={tags[tag].permalink} key={tag}>
            {tags[tag].name} ({tags[tag].count})
          </_Link.default>)}
        <hr />
      </article>).filter(item => item != null);
  return <_Layout.default title={title} wrapperClassName={_themeCommon.ThemeClassNames.wrapper.blogPages} pageClassName={_themeCommon.ThemeClassNames.page.blogTagsListPage} searchMetadatas={{
    // assign unique search tag to exclude this page from search results!
    tag: 'blog_tags_list'
  }}>
      <div className="container margin-vert--lg">
        <div className="row">
          <aside className="col col--3">
            <_BlogSidebar.default sidebar={sidebar} />
          </aside>
          <main className="col col--7">
            <h1>{title}</h1>
            <section className="margin-vert--lg">{tagsSection}</section>
          </main>
        </div>
      </div>
    </_Layout.default>;
}

var _default = BlogTagsListPage;
exports.default = _default;
